lib/WebDriver/Service/CurlService.php

```diff
@@ -87,21 +88,28 @@ public function execute($requestMethod, $url, $parameters = null, $extraOptions
         }
 
         curl_setopt($curl, CURLOPT_HTTPHEADER, $customHeaders);
-        curl_setopt($curl, CURLOPT_FAILONERROR, true);
 
         $rawResult = trim(curl_exec($curl));
         $info = curl_getinfo($curl);
+        // enrich the info a bit
+        $info['request_method'] = $requestMethod;
+
+        // NB: this only gets triggered when CURLOPT_FAILONERROR has been set in extraOptions.
+        // In that case, $rawResult is always empty.
+        if (CURLE_GOT_NOTHING !== ($errno = curl_errno($curl)) && $error = curl_error($curl)) {
+
+            curl_close($curl);
 
-        if (CURLE_GOT_NOTHING !== curl_errno($curl) && $error = curl_error($curl)) {
             $message = sprintf(
-                'Curl error thrown for http %s to %s%s',
+                "Curl error thrown for http %s to %s%s%s",
                 $requestMethod,
                 $url,
                 $parameters && is_array($parameters)
-                ? ' with params: ' . json_encode($parameters) : ''
+                ? ' with params: ' . json_encode($parameters) : '',
+                $error ? "\n\n$error" : ''
             );
-
-            throw WebDriverException::factory(WebDriverException::CURL_EXEC, $message . "\n\n" . $error);
+            $detailsException = new CurlDetailsException($error, $errno, null, $rawResult, $info);
```

Comment:
Why don't we throw this exception? The `CurlService` seems to be standalone component and isn't required to use `WebDriverException::factory` method.